### DISCUSSION TITLE
Is it possible to test different databaseUrls?

### DISCUSSION THREAD

**Discussion by yienchen97** [prescottprue/cypress-firebase#1312] [Q&A]
Hi, thanks for making this awesome lib!
It helps a lot in our testing.

I was wondering if this supports to test different dataBaseUrl?

We've tried to setup `https://test-a-url.firebaseio.com` as databaseUrl in firebase-admin in plugin/index.js, and init by below script with 

```js
 const firebaseConfig = {
   ...
   databaseURL: 'https://test-b-url.firebaseio.com',
   ...
 }

firebase.initializeApp(firebaseConfig)
attachCustomCommands({ Cypress, cy, firebase})
```

but while running cy.callRtdb, it still reads `https://test-a-url.firebaseio.com`.
I saw the "Make sure you use the same databaseURL when initializing the firebase instance within your app code", does it mean currently this lib only support single databaseURL? or there's any way to approch connecting multiple db urls in the tests? 

Thanks again for creating this lib!

**Comment by prescottprue** [comment]
Hey - thanks for reaching out with the question!

Calls to the database happen through tasks (on the node side) which are called by custom commands (browser side). Because of this, as mentioned in [the setup section in the README](https://github.com/prescottprue/cypress-firebase?tab=readme-ov-file#setup), you must pass `databaseURL` as additional config when calling `cypressFirebasePlugin` within `setupNodeEvents`:

In `cypress.config.js`:

```js
export default defineConfig({
  e2e: {
    baseUrl: 'http://localhost:3000',
    // NOTE: Add "supportFile" setting if separate location is used
    setupNodeEvents(on, config) {
      // e2e testing node events setup code
      return cypressFirebasePlugin(on, config, admin, {
        // Here is where you can pass special options.
        // if your databaseURL is not just your projectId plus ".firebaseio.com", then you _must_ give it here, like so:
        databaseURL: 'https://test-b-url.firebaseio.com',
      });
    },
  },
});
```
This is because internally, the [plugin initializes `firebase-admin` if it is not already initialized](https://github.com/prescottprue/cypress-firebase/blob/main/src/plugin.ts#L33) - it appears your example is referring to the client side Firebase SDK (i.e. `firebase` not `firebase-admin`). Both should be provided this database config -

Since the question came up, I'll try to think of ways to make it more clear within the README - [the "use different rtdb instance" section kinda touches on this, but that example is specific to emulator case and only different instances, not fully different DBs](https://github.com/prescottprue/cypress-firebase?tab=readme-ov-file#use-different-rtdb-instance). Open to ideas around this

**Reply by yienchen97** [reply]
Hi Scott, thanks so much for the reply!

I had set up databaseURL in both firebase-admin side and client side sdk:

in `cypress.config.js`
```js
export default defineConfig({
  e2e: {
    ...
    setupNodeEvents(on, config) {
      return cypressFirebasePlugin(on, config, admin, {
        databaseURL: 'https://test-a-url.firebaseio.com',
      });
    },
  },
});
```
in `commands.js`
```js
 const firebaseConfig = {
   ...
   databaseURL: 'https://test-b-url.firebaseio.com',
   ...
 }

firebase.initializeApp(firebaseConfig)
attachCustomCommands({ Cypress, cy, firebase})
```

Currently, by using the above settings, when I call `cy.callRtdb`, it still reads `https://test-a-url.firebaseio.com` as databaseURL instead of `https://test-b-url.firebaseio.com` that inited in the client side sdk.

I was wondering to know if this can be approached with this lib:
when cypress starts, the firebase admin config `https://test-a-url.firebaseio.com` is used as databaseURL, and after reading commands with initializeApp, `cy.callRtdb` reads `https://test-b-url.firebaseio.com` as the databaseURL set up in client side firebase sdk.
